lib/stat_tracker.rb

```diff
@@ -29,44 +29,29 @@ def lowest_total_score
   def biggest_blowout
     @game_collection.games.map {|game| (game.away_goals - game.home_goals).abs}.max
   end
-  # This only requires game information.
-  # It should probably move to game collection eventually.
-  def count_of_games_by_season
-    seasons = @game_collection.games.map{|game| game.season }.uniq  #season needs helper method
-    seasons.reduce({}) do |games_by_season, season|
-      games_per_season = @game_collection.games.find_all do |game|  #games_per_season needs helper method
-         season == game.season
-      end.length
-    games_by_season[season] = games_per_season
-    games_by_season
-    end
+
+                                                                    # This only requires game information.
+                                                                    # It should probably move to game collection eventually.
+  def count_of_games_by_season                                      #refactored by Ryan 2.22.20
+    games_by_season = @game_collection.all.group_by{|game| game.season}         #games_by_season 1st occurance
```

Comment:
I wonder if we should save games_by_season as a method in the GameCollection class. If we do this, we can remove one line of code which would be nice. But overall this looks goo!

Reply:
I think we should move it sometime in the future. we will still have to call upon it, but it will be so much cleaner.